File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
      int n = nums.length;
        int l = 0;
        int r = n - 1;
        int res = Integer.MAX_VALUE;

        while(l <= r){
          int m = l + ((r-l) >> 1);

          if(nums[m] >= nums[l]){
            res = Math.min(res , nums[l]);
            l = m +1;
          } else{
            res = Math.min(res , nums[m]);
            r = m - 1;
          }
        }

        return res;
    }
}

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1

        while l < r:
            m = l + (r-l)//2

            if nums[m] <= nums[r]: #Normal condition so go on the left
                r = m 
            else:
                l = m + 1 #Go on the left as we are closer to the rotatedt start
        
        return nums[l]
        